### DISCUSSION TITLE
How to survive flow deployment

### DISCUSSION THREAD

**Discussion by jfroy** [jensrossbach/node-red-contrib-chronos#101] [Q&A]
I am using delay nodes with dynamic delays and configuration messages in a flow. If those nodes get included in flow deployment (directly or indirectly) they lose their state. This is causing the flow to end up in a bad state. Are there tips and tricks for surviving deployments? I've looked at the schedule node source code as well, and I don't think it saves state for dynamic schedules.

**Comment by jensrossbach** [comment]
What about storing the state in an on-disk context variable? Then you could retrieve it again on a redeploy and pass it to the delay node (e.g., with an inject node). The state could also be retrieved directly from context variables: Using the custom delay type, you can use the JSONata functions `$flowContext`, `$globalContext` and `$env` to retrieve values from context or environment variables. Of course this is more complex and requires some knowledge of JSONata, but it is possible.
Also with a scheduler node, you can directly load dynamic data (in this case, the schedule) from such context variables (but this would not be suitable for delaying messages).
